ocaml/src/nsm_host_plugin.ml

```diff
@@ -375,6 +378,30 @@ let handle_query : type i o. read_user_db -> (i, o) Nsm_host_protocol.Protocol.q
         try Result.Ok (nsm_query tag namespace_id req)
         with Nsm_model.Err.Nsm_exn (err, _) -> Result.Error err)
        req
+  | UpdateSession ->
+     begin
+       let r =
+         List.fold_left
+         (fun acc (k,vo) ->
+           begin
+             match k with
+             | "manifest_ser" ->
+                begin
+                  match vo with
+                  | None -> acc
+                  | Some v ->
+                     let wanted_version = deserialize Llio.int8_from v in
+                     let new_version = min 2 wanted_version in
+                     let () =
+                       Nsm_protocol.Session.set_manifest_ser session new_version in
+                     (k, v) :: acc
```

Comment:
server side should return the actual new_version that it will use to serialize manifests? (and that can be used by the client)